include/tvm/task_dynamics/VelocityDamper.h

```diff
@@ -67,14 +44,77 @@ namespace tvm
     class TVM_DLLAPI VelocityDamper : public abstract::TaskDynamics
     {
     public:
+      /** A structure grouping the parameters of a velocity damper.
+        * \sa VelocityDamper.
+        */
+      class TVM_DLLAPI Config
+      {
+      public:
+        /**
+          * \param di interaction distance \f$d_i\f$. We need \f$ d_i > d_s \f$.
+          * \param ds safety distance \f$d_s\f$.
+          * \param xsi damping parameter \f$\xi \f$. If xsi = 0, the value will
+          * be computed automatically, otherwise, we need \f$\xi > 0\f$.
+          * In automatic mode, the value is recomputed each time the error value
+          * is at a distance to its bound lower or equal to \p \di with the
+          * formula \f$ \xi = -\dfrac{d_i - d_s}{d^k - d_s} \dot{d}^k +
+          * \xi_{\mathrm{off}} \f$.
+          * \param xsiOff offset \f$ \xi_{\mathrm{off}} \f$ used in the automatic
+          * computation of \f$\xi\f$. Used only in the case xsi=0.
+          */
+        Config(double di, double ds, double xsi, double xsiOff=0);
+
+        double di_;
+        double ds_;
+        double xsi_;
+        double xsiOff_;
+      };
+
+      /** A structure grouping the parameters for an anisotropic velocity damper.
+       * \sa VelocityDamper
+       */
+      class TVM_DLLAPI AnisotropicConfig
+      {
+      public:
+        /**
+          * \param di interaction distance \f$d_i\f$. We need \f$ d_i > d_s \f$.
+          * \param ds safety distance \f$d_s\f$.
+          * \param xsi damping parameter \f$\xi \f$. If xsi = 0, the value will
+          * be computed automatically, otherwise, we need \f$\xi > 0\f$.
+          * In automatic mode, the value is recomputed each time the error value
+          * is at a distance to its bound lower or equal to \p \di with the
+          * formula \f$ \xi = -\dfrac{d_i - d_s}{d^k - d_s} \dot{d}^k +
+          * \xi_{\mathrm{off}} \f$.
+          * \param xsiOff offset \f$ \xi_{\mathrm{off}} \f$ used in the automatic
+          * computation of \f$\xi\f$. Used only in the case xsi=0.
+          *
+          * All parameters must be of the same size
+          */
+        AnisotropicConfig(const VectorConstRef & di,
+                                        const VectorConstRef & ds,
+                                        const VectorConstRef & xsi,
+                                        const std::optional<VectorConstRef> & xsiOff = std::nullopt);
```

Comment:
You should be able to do without an `optional` here:  `const VectorConstRef& xsiOff = Eigen::VectorXd()` is working as well, without any dynamic memory allocation, iirc (note that `VectorConstRef` would be enough, but we use `const &` throughout the repo). You can then test if the size of the vector is `0` or not.
If you remove the `std::optional` don't forget to remove the header inclusion.

Reply:
This is what I wrote at first but I though the extra-semantic that `std::optional` carries is interesting and worth the small cost of it (and I don't think it incurs dynamic allocation?)